### PR TITLE
Apply transform to textures on android

### PR DESCRIPTION
**Pull request by sigurdm**

### CODE REVIEW

File: shell/platform/android/android_external_texture_gl.cc [hunk 1]
```diff
@@ -44,16 +43,44 @@ void AndroidExternalTextureGL::Paint(SkCanvas& canvas, const SkRect& bounds) {
...
   GrGLTextureInfo textureInfo = {GL_TEXTURE_EXTERNAL_OES, texture_name_};
-  GrBackendTexture backendTexture(bounds.width(), bounds.height(),
-                                  kRGBA_8888_GrPixelConfig, textureInfo);
+  GrBackendTexture backendTexture(1, 1, kRGBA_8888_GrPixelConfig, textureInfo);
   sk_sp<SkImage> image = SkImage::MakeFromTexture(
       canvas.getGrContext(), backendTexture, kTopLeft_GrSurfaceOrigin,
       SkAlphaType::kPremul_SkAlphaType, nullptr);
   if (image) {
-    canvas.drawImage(image, bounds.x(), bounds.y());
+    SkMatrix originalMatrix(canvas.getTotalMatrix());
```
Comment (chinmaygarde) [inline shell/platform/android/android_external_texture_gl.cc]: You could replace the calls to save the original matrix and restore it with `SkCanvas.save` and `SkCanvas.restore`. There RAII version for this is `SkAutoCanvasRestore`.

File: shell/platform/android/android_external_texture_gl.cc [hunk 2]
```diff
@@ -44,16 +43,44 @@ void AndroidExternalTextureGL::Paint(SkCanvas& canvas, const SkRect& bounds) {
     new_frame_ready_ = false;
   }
   GrGLTextureInfo textureInfo = {GL_TEXTURE_EXTERNAL_OES, texture_name_};
-  GrBackendTexture backendTexture(bounds.width(), bounds.height(),
-                                  kRGBA_8888_GrPixelConfig, textureInfo);
+  GrBackendTexture backendTexture(1, 1, kRGBA_8888_GrPixelConfig, textureInfo);
```
Comment (chinmaygarde) [inline shell/platform/android/android_external_texture_gl.cc]: Hah. Thats clever!

Comment (sigurdm) [reply shell/platform/android/android_external_texture_gl.cc]: :)

File: shell/platform/android/android_external_texture_gl.cc (hunk 1 above)
Comment (sigurdm) [reply shell/platform/android/android_external_texture_gl.cc]: Done

**Pull request merged**